runtime/vm/visible.c

```diff
@@ -343,6 +355,15 @@ loadAndVerifyNestHost(J9VMThread *vmThread, J9Class *clazz, UDATA options)
 		/* If a problem occurred in nest host verification then the nest host value is invalid */
 		if (J9_VISIBILITY_ALLOWED == result) {
 			clazz->nestHost = nestHost;
+		} else if (hiddenNestMate) {
+			/* 
+			 * For hidden class that is added into the nest of its host class, i.e. defined with Option.NESTMATE,
+			 * if we cannot get its nest host, its hostclass (curClazz) shuold be used as nest host instead. 
+			 */
+			clazz->nestHost = curClazz;
```

Comment:
It might be clearer to directly assign `clazz->nestHost = clazz->hostClass;` here as `curClazz` is updated in the `while` loop above

Reply:
clazz->hostClass could be another hidden class defined with Option.NESTMATE, so we need to walk to the end of the chain to use the class that is not hidden NestMate class, which is curClazz.

Reply:
I thought we discussed in another PR that a hidden class couldn't be the host of another hidden class... or am I remembering that wrong?

Reply:
That discussion was a hidden class cannot be super class of another hidden class.